Fix automatic window resizing in i3

Without this patch, the window would become tiny every time the virtual machine
changes its screen resolution.

--- VirtualBox-7.0.4.orig/src/VBox/Frontends/VirtualBox/src/globals/UIDesktopWidgetWatchdog.cpp
+++ VirtualBox-7.0.4/src/VBox/Frontends/VirtualBox/src/globals/UIDesktopWidgetWatchdog.cpp
@@ -443,7 +443,7 @@
     /* Get cached available-geometry: */
     const QRect availableGeometry = m_availableGeometryData.value(iHostScreenIndex);
     /* Return cached available-geometry if it's valid or screen-geometry otherwise: */
-    return availableGeometry.isValid() ? availableGeometry :
+    return availableGeometry.isValid() && NativeWindowSubsystem::X11WindowManagerType() != X11WMType_i3 ? availableGeometry :
            QApplication::desktop()->screenGeometry(iHostScreenIndex);
 #  endif /* !VBOX_GUI_WITH_CUSTOMIZATIONS1 */
 # else /* !VBOX_WS_X11 */
--- VirtualBox-7.0.4.orig/src/VBox/Frontends/VirtualBox/src/platform/x11/VBoxUtils-x11.cpp
+++ VirtualBox-7.0.4/src/VBox/Frontends/VirtualBox/src/platform/x11/VBoxUtils-x11.cpp
@@ -117,6 +117,9 @@
                 else
                 if (QString((const char*)pcData).contains("Xfwm4", Qt::CaseInsensitive))
                     wmType = X11WMType_Xfwm4;
+                else
+                if (QString((const char*)pcData).contains("i3", Qt::CaseInsensitive))
+                    wmType = X11WMType_i3;
                 if (pcData)
                     XFree(pcData);
             }
--- VirtualBox-7.0.4.orig/src/VBox/Frontends/VirtualBox/src/platform/x11/VBoxUtils-x11.h
+++ VirtualBox-7.0.4/src/VBox/Frontends/VirtualBox/src/platform/x11/VBoxUtils-x11.h
@@ -49,6 +49,7 @@
     X11WMType_Metacity,
     X11WMType_Mutter,
     X11WMType_Xfwm4,
+    X11WMType_i3,
 };
 
 /** X11: Screen-saver inhibit methods. */
@@ -95,7 +96,7 @@
     /** X11: Determines and returns whether the compositing manager is running. */
     bool X11IsCompositingManagerRunning();
     /** X11: Determines and returns current Window Manager type. */
-    X11WMType X11WindowManagerType();
+    SHARED_LIBRARY_STUFF X11WMType X11WindowManagerType();
 
 #if 0 // unused for now?
     /** X11: Inits the screen saver save/restore mechanism. */
--- VirtualBox-7.0.4.orig/src/VBox/Frontends/VirtualBox/src/widgets/UIMiniToolBar.cpp
+++ VirtualBox-7.0.4/src/VBox/Frontends/VirtualBox/src/widgets/UIMiniToolBar.cpp
@@ -888,6 +888,10 @@
     /* Enable translucency through Qt API if supported: */
     if (uiCommon().isCompositingManagerRunning())
         setAttribute(Qt::WA_TranslucentBackground);
+
+    if (NativeWindowSubsystem::X11WindowManagerType() == X11WMType_i3)
+        setWindowFlags(Qt::FramelessWindowHint);
+
 #endif /* VBOX_WS_X11 */
 
     /* Make sure we have no focus: */
